File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java

import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0; i<nums.length;i++){
            map.put(nums[i], map.getOrDefault(nums[i], 1)+1);
        }
        
        for(int key: map.keySet()){
            System.out.println(key);
            answer+=1;
        }
        
        answer = Math.min(answer, nums.length/2);
        
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> nums)
{
    unordered_map<int,int> map;
    
    for(int i=0; i<nums.size();i++){
        map[nums[i]]++;
    }
    
    int answer = min(nums.size()/2 , map.size());
    
    return answer;
}